### PR TITLE
(DOCS-9378) Add information for Apps Datastore feature

### PR DESCRIPTION
**Pull request by neko-dd**
### What does this PR do? What is the motivation?

Adds datastore feature to Workflows and App Builder

### Merge instructions

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/service_management/workflows/datastore.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- link: "service_management/workflows/build"
+  tag: "Documentation"
+  text: "Build Workflows"
+---
+
+{{< callout url="https://docs.google.com/forms/d/1gVVWLg5P3EXkVXBc4qZtqKp_nqykDsFXCMmh6pbvxx0/edit" btn_hidden="false" header="Try the Preview!">}}
+Datastore is in Preview. Use this form to request access today.
+{{< /callout >}}
+
+## Overview
```
Comment (barakshush) [inline content/en/service_management/workflows/datastore.md:17]: - I think we should emphasis that you can interact with the datastore from an app or from a WF. Currently, we only say: "structured data storage solution within Datadog's App Builder and Workflow Automation products" Do you think it is clear enough?
- It is very important to mention that the Datastore has also a UI layer that allow users to interact with the table from the UI. This is our advantage over solutions like S3 or DynamoDB

File: content/en/service_management/workflows/datastore.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+The Apps Datastore offers a scalable, structured data storage solution within Datadog's App Builder and Workflow Automation products. It supports CRUD (Create, Read, Update, and Delete) operations and integrates seamlessly with Datadog's ecosystem to optimize persistent data storage without the need for external databases.
+
+## Create a datastore
```
Comment (barakshush) [inline content/en/service_management/workflows/datastore.md:21]: I see that you listed here:
1. Create a datastore
2. Edit a datastore
3. Reference a datastore

The only thing that is missing here for me is "Deleting a Datastore"

File: content/en/service_management/workflows/datastore.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
```
Comment (barakshush) [inline content/en/service_management/workflows/datastore.md:22]: A few general comments:
1. Do you think it is also important to mention that it is possible to search the table using the search field?
2.Do you think it is also important to mention that it is possible sort the table by every column by clicking on the column name?
3. I think it is also important to explain how to interact with the datastore from an App / WF and list the relevant actions

Comment (neko-dd) [reply content/en/service_management/workflows/datastore.md:22]: 1. I'll take a look.
2. This is already explained in the "reference" section.

File: content/en/service_management/workflows/datastore.md (hunk 1 above)
Comment (neko-dd) [reply content/en/service_management/workflows/datastore.md:17]: Added some things to the intro

File: content/en/service_management/workflows/datastore.md (hunk 2 above)
Comment (neko-dd) [reply content/en/service_management/workflows/datastore.md:23]: Added a delete section

File: content/en/service_management/workflows/datastore.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  text: "Build Apps"
+- link: "service_management/workflows/build"
+  tag: "Documentation"
+  text: "Build Workflows"
+---
+
+{{< callout url="https://docs.google.com/forms/d/1gVVWLg5P3EXkVXBc4qZtqKp_nqykDsFXCMmh6pbvxx0/edit" btn_hidden="false" header="Try the Preview!">}}
+Datastore is in Preview. Use this form to request access today.
+{{< /callout >}}
+
```
Comment (pingxia23) [inline content/en/service_management/workflows/datastore.md:16]: Could we add a section about the permissions required? https://datadoghq.atlassian.net/wiki/spaces/APPS/pages/4206395778/How+to+Use+Apps+Datastore#Setup-the-permissions

Comment (neko-dd) [reply content/en/service_management/workflows/datastore.md:16]: The permissions listed there are what your Datadog account needs to have, is that correct?

Comment (neko-dd) [reply content/en/service_management/workflows/datastore.md:16]: Added a section.

File: content/en/service_management/workflows/datastore.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+## Overview
+
+The Apps Datastore offers a scalable, structured data storage solution within Datadog's App Builder and Workflow Automation products. It supports CRUD (Create, Read, Update, and Delete) operations and integrates seamlessly with Datadog's ecosystem to optimize persistent data storage without the need for external databases.
+
+You can interact with a datastore using an app or a workflow, or you can use the UI in the Datadog App.
+
+
+## Prerequisites
+
+To interact with the Apps Datastore, your Datadog account must have the following [permissions][6], all of which are included in the Datadog Standard Role:
```
Comment (pingxia23) [inline content/en/service_management/workflows/datastore.md:26]: To clarify
* if you need to access the datastore page, you need all permissions
* if you only need to use datastore from app builder and/or workflow, you don't need `apps_datastore_manage` permission

**Review by pingxia23 — COMMENTED** [review]
One nit comment

otherwise looks good to me 

I'll defer to other reviewer for final approval

File: content/en/service_management/workflows/datastore.md (hunk 5 above)
Comment (neko-dd) [reply content/en/service_management/workflows/datastore.md:26]: Ok, thanks! I will update

File: content/en/service_management/workflows/datastore.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+
+## Prerequisites
+
+To interact with the Apps Datastore, your Datadog account must have the following [permissions][6], which are included in the Datadog Standard Role:
+
+* `apps_datastore_read` - Allows read access to the data within the Apps Datastore.
+* `apps_datastore_write` - Allows modification of data within the Apps Datastore, including adding, editing, and deleting records.
+
+To use the use the [Apps Datastore UI][1], you also need the following permission:
```
Comment (pingxia23) [inline content/en/service_management/workflows/datastore.md:31]: This permission is also included in Datadog Standard Role

**Review by pingxia23 — APPROVED** [review]

File: content/en/service_management/workflows/datastore.md (hunk 6 above)
Comment (neko-dd) [reply content/en/service_management/workflows/datastore.md]: Just added that as you were giving this feedback! Thanks for catching that :)

**Comment by buraizu** [comment]
Created DOCS-9448 to review

**Review by aliciascott — APPROVED** [review]
looks good

**Review by aliciascott — APPROVED** [review]
that looks great!

**Pull request merged**